apps/eew/scfinder/main.cpp

```diff
@@ -480,6 +503,144 @@ class App : public Client::StreamApplication {
+				Origin *org = Origin::Cast(it->get());
+				if ( org )
+					addObject("", org);
+			}
+		}
+
+		void addObject(const std::string &parentID, Object *obj) {
+			
+			Origin *org = Origin::Cast(obj);
+			if ( _processOrigins ) {
+				if ( org ) {
+					logObject(_inputOrgs, Core::Time::GMT());
+					try {
+						Core::Time orgTime = Core::Time::GMT();
+						SEISCOMP_DEBUG("Received %s origin from %s (lat: %f km lon:%f km  dep: %f) wrapped in origin: %s at %s", 
+								org->methodID().c_str(),
+								org->creationInfo().author().c_str(),
+								org->latitude().value(),
+								org->longitude().value(),
+								org->depth().value(),
+								org->publicID().c_str(),
+								orgTime.iso().c_str()); 
+
+						if ( strcmp(org->methodID().c_str() , "FinDer") == 0 ) {
+							SEISCOMP_DEBUG("Discarding origin %s", 
+											org->publicID().c_str());
+							return;
+						}
+						double   epicentralPGAvalue = _defaultEpPGA;
+
+						if ( _defaultEpPGA > 0 ) {
+							SEISCOMP_DEBUG("Using default epicentral PGA of %f cm/s/s", 
+											_defaultEpPGA);
+						} else {
+							// percentile(PGAs, 95)*1.2 or fit line to PGA=ft(dist,cutoff dist) 
+							for ( size_t i = 0; i < _latestMaxPGAs.size(); ++i ) {
+								PGA_Data &pga = _latestMaxPGAs[i];
+								if ( pga.get_value() > epicentralPGAvalue ){
+									epicentralPGAvalue = pga.get_value();
+								}
+							}
+							//epicentralPGAvalue *= 2.0 ;	
+						}
+						
+						int neppgas = 0;		
+						for ( int i=-1*_nPtsPGA;i<(_nPtsPGA+1);i++ ) {
+							for ( int j=-1*_nPtsPGA;j<(_nPtsPGA+1);j++ ) {
+								_latestMaxPGAs.push_back(
+									PGA_Data(
+											"EP"+std::to_string(neppgas),
+											"X",
+											"XXX",
+											"XX",
+											Coordinate( org->latitude().value()+0.01*i,
+														org->longitude().value()+0.01*j ),
+											(double)epicentralPGAvalue/(1+pow(pow(i,2)+pow(j,2),0.5)),
+											orgTime.seconds() 
+										)
+									);
+								neppgas++ ;
+							}
+						}
+
+						for ( size_t i = 0; i < _latestMaxPGAs.size(); ++i ) {
+							PGA_Data &pga = _latestMaxPGAs[i];
+							if ( strcmp( pga.get_network().c_str(), "X" ) != 0 ) {
+								continue;
+							}
+							SEISCOMP_DEBUG("%s.%s.%s.%s %f %f %s %f",
+											pga.get_network().c_str(),
+											pga.get_name().c_str(),
+											pga.get_location_code().c_str(),
+											pga.get_channel().c_str(),
+											pga.get_location().get_lat(),
+											pga.get_location().get_lon(),
+											Core::Time(pga.get_timestamp()).iso().c_str(),
+											pga.get_value());
+						}
+						SEISCOMP_DEBUG("epicentral PGA value: %f cm/s/s",epicentralPGAvalue);	
+
+						// get the current time for this new finder object
+						long event_id = (long)time(NULL);
+						
+						// make constructor take a Coordinate designated by *cit...it will increment Nfinder
+						_finderList.push_back(new Finder(Coordinate(org->latitude().value(), 
```

Comment:
Use my own separate local FinDer() object directly with .process() (it's a pointer)